lib/matplotlib/cbook.py

```diff
@@ -492,17 +504,23 @@ def connect(self, s, func):
         func will be called
         """
         self._func_cid_map.setdefault(s, WeakKeyDictionary())
-        if func in self._func_cid_map[s]:
```

Comment:
@WeatherGod This conversation thread is where the bug is.  The issue is that it does not catch that the same function has gone in more than once.